source/Mlos.Python/mlos/OptimizerEvaluationTools/OptimizerEvaluationReport.py

```diff
@@ -0,0 +1,134 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+import json
+import os
+import pickle
+from typing import Dict, List
+from mlos.Logger import BufferingHandler, create_logger
+from mlos.OptimizerEvaluationTools.OptimumOverTime import OptimumOverTime
+from mlos.Optimizers.RegressionModels.RegressionModelFitState import RegressionModelFitState
+from mlos.Spaces import Point
+from mlos.Tracer import trace, Tracer
+
+
+class OptimizerEvaluationReport:
+    """Contains all information gathered during an optimizer evaluation run.
+
+    This includes:
+        * optimizer configuration
+        * objective function configuration
+        * serialized optimizer (with random seeds, and all observations)
+        * serialized objective function (with random seeds)
+        * evaluation parameters:
+            * num optimization iterations
+            * evaluation frequency
+        * optimizer's regression model goodness of fit metrics over time
+        * optima over time for the following definitions:
+            * best observation
+            * best predicted value for observed config
+            * best upper confidence bound on a 99% confidence interval for an observed config
+            * best lower confidence bound on a 99% confidence interval for an observed config
+        * execution trace as captured by the mlos.Tracer

```

Comment:
Add a work item to also include the logs.

Add another work item to add streaming logs and traces directly to disk instead of buffering them in memory.

Reply:
#158